lib/DevServerManager.ts

```diff
@@ -1,48 +1,86 @@
-const { DevModeInterface } = require('@hubspot/ui-extensions-dev-server');
-const {
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { Environment } from '@hubspot/local-dev-lib/types/Config';
+import {
+  COMPONENT_TYPES,
+  ComponentTypes,
+  Component,
+} from './projects/structure';
+import { i18n } from './lang';
+import { promptUser } from './prompts/promptUtils';
+import { DevModeInterface as UIEDevModeInterface } from '@hubspot/ui-extensions-dev-server';
+import {
   startPortManagerServer,
   stopPortManagerServer,
   requestPorts,
-} = require('@hubspot/local-dev-lib/portManager');
-const {
+} from '@hubspot/local-dev-lib/portManager';
+import {
   getHubSpotApiOrigin,
   getHubSpotWebsiteOrigin,
-} = require('@hubspot/local-dev-lib/urls');
-const { getAccountConfig } = require('@hubspot/local-dev-lib/config');
+} from '@hubspot/local-dev-lib/urls';
+import { getAccountConfig } from '@hubspot/local-dev-lib/config';
+import { ProjectConfig } from '../types/Projects';
 
 const i18nKey = 'lib.DevServerManager';
 
 const SERVER_KEYS = {
   privateApp: 'privateApp',
   publicApp: 'publicApp',
+} as const;
+
+type ServerKey = keyof typeof SERVER_KEYS;
+
+type DevServerInterface = {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
+  setup?: Function;
+  start?: (options: object) => Promise<void>;
+  fileChange?: (filePath: string, event: string) => Promise<void>;
+  cleanup?: () => Promise<void>;
+};
+
+type DevServer = {
+  componentType: ComponentTypes;
+  serverInterface: DevServerInterface;
+};
+
+type ComponentsByType = {
+  [key in ComponentTypes]: { [key: string]: Component };
 };
 
 class DevServerManager {
+  private initialized: boolean;
+  private started: boolean;
+  private componentsByType: ComponentsByType;
+  private devServers: { [key in ServerKey]: DevServer };
+
   constructor() {
     this.initialized = false;
     this.started = false;
-    this.componentsByType = {};
-    this.server = null;
-    this.path = null;
+    this.componentsByType = {
+      [COMPONENT_TYPES.privateApp]: {},
+      [COMPONENT_TYPES.publicApp]: {},
+      [COMPONENT_TYPES.hublTheme]: {},
+    };
     this.devServers = {
       [SERVER_KEYS.privateApp]: {
         componentType: COMPONENT_TYPES.privateApp,
-        serverInterface: DevModeInterface,
+        serverInterface: UIEDevModeInterface,
       },
       [SERVER_KEYS.publicApp]: {
         componentType: COMPONENT_TYPES.publicApp,
-        serverInterface: DevModeInterface,
+        serverInterface: UIEDevModeInterface,
       },
     };
   }
 
-  async iterateDevServers(callback) {
-    const serverKeys = Object.keys(this.devServers);
+  async iterateDevServers(
+    callback: (
+      serverInterface: DevServerInterface,
+      compatibleComponents: {
+        [key: string]: Component;
+      }
+    ) => Promise<void>
+  ): Promise<void> {
+    const serverKeys: ServerKey[] = Object.keys(this.devServers) as ServerKey[];
```

Comment:
Is there a better way to do this without having to type cast?

Reply:
Unfortunately there is not